substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -713,6 +713,18 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::add_member_to_rank(*rank as u32))]
+		pub fn add_member_to_rank(
+			origin: OriginFor<T>,
+			who: AccountIdLookupOf<T>,
+			rank: Rank,
+		) -> DispatchResult {
```

Comment:
Won't we need a similar call in `core-fellowship`? Or will this somehow bypass that configuration?

Reply:
Don't know. @ggwpez could you please take a look at PR and maybe help to answer this question. Thank you)

Reply:
I think we should have only one in the core-fellowship pallet. Similar to the `induct` function there.

Reply:
@ggwpez  few questions:
1. Does it make sense to make another call similar to `promote`?
2. Maybe it's better to close #262 if the issue task is no longer valid?

Reply:
The task is definitely valid, it just does not prescribe a solution.

A `promote_to` function could also be useful for people already inducted but where it makes sense for some reason to promote them over several ranks.

Reply:
> The task is definitely valid, it just does not prescribe a solution.
> 
> A `promote_to` function could also be useful for people already inducted but where it makes sense for some reason to promote them over several ranks.

But this violate the rule of mainefest 4.2.3: "...associated rank is incremented by one". It suppose at least from the manifest it is not possible to increment rank more than 1 at a time. Or am i missing something?

Reply:
Two things:

One, the manifesto is from the Tech Fellowship. It is not the only collective (see, e.g., the Ambassador collective, which actually requires promoting members by 3 steps). If the Fellowship never wants someone promoted by more than one rank at a time they will need to configure things as such.

Two, we use origins to control access to privileged functions, and typically origins that correspond to public tracks will always outweigh any internal management origins.

What I mean is that rules like that are mainly arbitrary and can always be superseded by governance. You could also do a runtime upgrade or referendum to `set_storage` to get the same effect. The goal of this task is to provide a transparent, non-footgun way of accomplishing this effect.

Things like manifestos and white papers are guiding principles but they shouldn't trump pragmatism. The JAM prize, which the Fellowship also agreed to steward, has conditions like automatic promotion to X rank for certain accomplishments. Sure, the manifesto has that clause and it can be upheld by the members and enforced by origin configuration, but there are obvious pragmatic shortfalls of the code in some circumstances that led to the creation of this feature request.